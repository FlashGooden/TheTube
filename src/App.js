import React from "react";
import "./App.scss";
// import {HeaderNav} from './containers/HeaderNav/HeaderNav'
// import {SideBar} from './containers/SideBar/SideBar'
import { AppLayout } from "./components/AppLayout/AppLayout";
import { Home } from "./containers/Home/Home";

export function App() {
  return (
    <div>
      <AppLayout>
        <Home />
      </AppLayout>
    </div>
  );
}

export default App;
