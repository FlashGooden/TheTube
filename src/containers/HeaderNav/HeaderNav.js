import React, { Component } from "react";
import { Image, Menu, Form, Input } from "semantic-ui-react";
import "./HeaderNav.scss";
import logo from "../../assets/images/TheTube.png";

export default class HeaderNav extends Component {


  render() {
    return (
      <>
        <Menu borderless className="top-menu" fixed="top">
          <Menu.Item header className="logo">
            <Image src={logo} size="tiny" />
          </Menu.Item>
          <Menu.Menu className="nav-container">
            <Menu.Item className="search-input">
              <Form>
                <Form.Field>
                <Input placeHolder="search"/>
              </Form.Field>
              </Form>
              
            </Menu.Item>
          </Menu.Menu>
        </Menu>
      </>
    );
  }
}


