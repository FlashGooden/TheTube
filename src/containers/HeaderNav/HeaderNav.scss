@import '../../styles/shared.scss';

.ui.menu.top-menu {
  border: none;
  height: $header-nav-height;
  .logo {
    width: $sidebar-left-width;
  }
  .nav-container {
    justify-content: space-between;
    flex-grow: 1;
    padding: 0;
    .search-input {
      padding-left: 0;
      width: 33%;
      form{
        width: 100%;
      }
    }
  }
  .header-icon {
    color: #DE003C;
  }
}
