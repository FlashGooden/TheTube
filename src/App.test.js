import { render } from '@testing-library/react';
import {HeaderNav} from './containers/HeaderNav/HeaderNav'
import {SideBar} from './containers/SideBar/SideBar'
import { create } from "react-test-renderer";
import React from 'react';
import {shallow} from 'enzyme'
import {App} from './App';
import assert from 'assert'

test('Renders App component', () => {
  const wrapper = shallow(
    <App />
  )
  expect(wrapper).toMatchSnapshot();
});

class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = { text: "" };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(() => {
      return { text: "PROCEED TO CHECKOUT" };
    });
  }

  render() {
    return (
      <button onClick={this.handleClick}>
        {this.state.text || this.props.text}
      </button>
    );
  }
}

describe("Button component", () => {
  test("it shows the expected text when clicked (testing the wrong way!)", () => {
    const component = create(<Button text="SUBSCRIBE TO BASIC" />);
    const instance = component.root;
    console.log( component.root.props)
    const button = instance.findByType("button");
    button.props.onClick();
    expect(button.props.children).toBe("PROCEED TO CHECKOUT");
  });
});

assert.deepEqual({'hi': "no"},{'hi': 'no'})