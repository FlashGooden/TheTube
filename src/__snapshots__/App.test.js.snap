// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App component 1`] = `
<div>
  <AppLayout>
    <Home />
  </AppLayout>
</div>
`;
